common/lib/xmodule/xmodule/modulestore/tests/test_assetstore.py

```diff
@@ -423,13 +424,53 @@ def check_asset_values(assets, orig):
 
                 # Check 'em.
                 for asset_type, asset_list in (
-                    ('different', differents),
-                    ('vrml', vrmls),
-                    ('asset', regular_assets),
+                    ('different', self.differents),
+                    ('vrml', self.vrmls),
+                    ('asset', self.regular_assets),
+                ):
+                    assets = store.get_all_asset_metadata(course.id, asset_type)
+                    self.assertEquals(len(assets), len(asset_list))
+                    self._check_asset_values(assets, asset_list)
+
+                self.assertEquals(len(store.get_all_asset_metadata(course.id, 'not_here')), 0)
+                self.assertEquals(len(store.get_all_asset_metadata(course.id, None)), 4)
+
+                assets = store.get_all_asset_metadata(
+                    course.id, None, start=0, maxresults=-1,
```

Comment:
`maxresults=-1` means return all results. I copied the pattern from the contentstore in content.py - trying to keep things consistent.

I'm not quite following what you mean about `sort`. I'm again following the contentstore pattern here. How would it change? Would I break out the `sort` parameter into two separate arguments?